### DISCUSSION TITLE
offline parsing binlog support for mysql 5.7.23

### DISCUSSION THREAD

**Discussion by rgyu** [rusuly/MySqlCdc#25] [Q&A] [answered]
I tried to run the sample program with mysql 5.7.23 binlog file, but seems cannot print.  The code piece is exactly the same as sample.  Better to have a sample binlog file attached to the project, so that I can confirm whether it's because of binlog file format issue or other reasons. 

            string binlogfile = @"D:\incoming\GGG-bin.000002";
            
            using (Stream stream = File.OpenRead(binlogfile))
            {
                EventDeserializer deserializer = new MySqlEventDeserializer();
                var reader = new BinlogReader(deserializer, stream);              
                await foreach (var binlogEvent in reader.ReadEvents())
                {                    
                    await PrintEventAsync(binlogEvent);
                }
            }

**Comment by rusuly** [comment, marked as answer]
Hi,
I installed MySQL 5.7.23, enabled binary logging and generated three inserts.
Everything seems to work fine. You can check the binlog file that I uploaded to the source code.
I replaced the Main method code with the the line
```csharp
static async Task Main(string[] args)
{
    await OfflineSample.Start(false);
}
```
<img width="620" alt="Screenshot 2021-09-12 at 21 16 14" src="https://user-images.githubusercontent.com/31304442/132998369-58c9fe2f-56a8-43b1-b48c-8d87eeac13c3.png">
